server.rb

```diff
@@ -11,14 +11,25 @@ def fixnum_param_with_default(key, default)
   param_with_default(key, default).to_i
 end
 
+def color_param_with_default(key, default)
+  if params[key] === "transparent" 
+   return params[key]
+  elsif /^([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/ === params[key] 
+    return "##{params[key]}"
+  else
+    return "##{default}"
+  end
```

Comment:
A few things here:

* Mind taking out the explicit `return`s? They [aren't needed](https://github.com/bbatsov/ruby-style-guide#no-explicit-return).
* I think the regular expression is a bit easier to read as `/^([a-f0-9]{6}|[a-f0-9]{3})$/i`.
* Can we make it clear that the regular expression is looking for a hexidecimal color?

Overall, I think it can be simplified to:

```ruby
def hex_color?(val)
  /^([a-f0-9]{6}|[a-f0-9]{3})$/i =~ val
end

def convert_color_param(val)
  if hex_color?(val)
    "##{val}"
  else
    val
  end
end

background_color = param_with_default(:background_color, 'FFFFFF')
background_color = convert_color_param(background_color)
```

I don't love the name of the `convert_color_param` (open to ideas), but how does the rest sound?